geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/Customer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.lucene.test;
+
+import java.io.Serializable;
+
+public class Customer implements Serializable {
+  private String name;
+  private Person contact; // search nested object
+
+  public Customer(String name) {
+    this.name = name;
+
+    // purposely use Tommi instead of Tommy, Johnni instead of Johnny in contact.name
+    this.contact = new Person(name.replace('y', 'i'));
```

Comment:
I don't think test objects should have this kind of logic in them. The test itself should figure out what data is held in the customer and just pass it in. As it stands someone reading the test has no idea that there are string substitutions happening down inside Customer.